src/data.rs

```diff
@@ -1067,6 +1074,12 @@ unsafe impl Sync for DataTreeOwningRef<'_> {}
 // ===== impl DataNodeRef =====
 
 impl<'a> DataNodeRef<'a> {
+    /// Returns a mutable raw pointer to the underlying C library representation
+    /// of the data node reference.
+    pub fn into_raw(self) -> *mut ffi::lyd_node {
```

Comment:
This could be `as_raw` as well since `DataNodeRef` doesn't own data (it doesn't have a `Drop` implementation, for instance). Other than that, the changes look great to me.